Documentation/gettingstarted/k8s-install-openshift-okd.rst

```diff
@@ -0,0 +1,202 @@
+
+Create an OpenShift OKD Cluster
+===============================
+
+First, set cluster name:
+
+.. code:: bash
+
+   CLUSTER_NAME="cluster-1"
+
+Now, create configuration files:
+
+   NOTE: the sample output below is showing the AWS provider, but
+   it should work the same way with other providers
+
+.. code:: bash
+
+   $ openshift-install create install-config --dir "${CLUSTER_NAME}"
+   ? SSH Public Key ~/.ssh/id_rsa.pub
+   ? Platform aws
+   INFO Credentials loaded from default AWS environment variables 
+   ? Region eu-west-1
+   ? Base Domain openshift-test-1.cilium.rocks
+   ? Cluster Name cluster-1
+   ? Pull Secret [? for help] ******************************************************************************************************************************************************************************************
+
+And set ``networkType: Cilium``:
+
+.. code:: bash
+
+   sed -i 's/networkType:\ OVNKubernetes/networkType:\ Cilium/' "${CLUSTER_NAME}/install-config.yaml"
+
+Next, generate OpenShift manifests:
+
+.. code:: bash
+
+   openshift-install create manifests --dir "${CLUSTER_NAME}"
+
+Now, define ``cilium`` namespace:
+
+.. code:: bash
+
+   cat << EOF > "${CLUSTER_NAME}/manifests/cluster-network-03-cilium-namespace.yaml"
+   apiVersion: v1
+   kind: Namespace
+   metadata:
+     name: cilium
+     annotations:
+       # node selector is required to make cilium-operator run on control plane nodes
+       openshift.io/node-selector: ""
+     labels:
+       name: cilium
+       # run level sets priority for Cilium to be deployed prior to other components
+       openshift.io/run-level: "0"
+   EOF
+
+.. include:: k8s-install-download-release.rst
+
+Next, render Cilium manifest:
+
+.. parsed-literal::
+
+   helm template |CHART_RELEASE|  \\
+      --namespace cilium \\
+      --set config.ipam=cluster-pool \\
+      --set global.cni.binPath=/var/lib/cni/bin \\
+      --set global.cni.confPath=/var/run/multus/cni/net.d \\
+      --set global.ipam.operator.clusterPoolIPv4PodCIDR=10.128.0.0/14 \\
+      --set global.ipam.operator.clusterPoolIPv4MaskSize=23 \\
+      --set global.nativeRoutingCIDR=10.128.0.0/14 \\
+      --set config.bpfMasquerade=false \\
+      --set global.endpointRoutes.enabled=true \\
+      --output-dir "${OLDPWD}"
+   cd "${OLDPWD}"
+
+.. note:: If you make any changes to CIDRs in ``"${CLUSTER_NAME}/install-config.yaml"`` you need to make
+   sure that ``clusterPoolIPv4PodCIDR`` and ``clusterPoolIPv4MaskSize`` reflect those changes.
+
+And copy Cilium manifest to ``${CLUSTER_NAME}/manifests``:
+
+.. code:: bash
+
+   for component in config agent operator
+      do for resource in cilium/charts/${component}/templates/*
+          do cp "${resource}" "${CLUSTER_NAME}/manifests/cluster-network-04-cilium-${component}-$(basename ${resource})"
+      done
+   done
+
+Create the cluster:
+
+.. note:: The sample output below is showing the AWS provider, but
+   it should work the same way with other providers.
```

Comment:
Formatting is a little unusual on this one, but I don't check to see whether it renders correctly.

Reply:
This is the second time this note is present, I wonder if it'd be better to just mention at the top of the doc that these instructions will demonstrate command output for an AWS environment?

Reply:
It appears to render fine, but I should probably double-check. It is duplicated somewhat intentionally, I think it's important to keen the note next to the output, as it only relates to the to output samples. Maybe we can put the text in a variable or an include file?

Reply:
I'm not too worried then if you think it's a good idea to call it out for each snippet.